inc/zoo/swar/associative_iteration.h

```diff
@@ -205,6 +210,34 @@ constexpr auto associativeOperatorIterated_regressive(
     return result;
 }
 
+
```

Comment:
@thecppzoo It would be useful to get your feedback on this example. Been scratching my head to understand this, and I think I might need a nudge for it to click. I think it's the `forSquaring` which is still confusing to me. 

I think for me being able to generalise between a vanilla type and a SWAR type using the same template would really help me understand the concept more firmly.